### PR TITLE
Assertion: Table should have following values

### PR DESCRIPTION
**Pull request by jose-ochoa**
Using the header columns to assert that a table contains a row with the provided values, converts the following madness:

```Gherkin
Then the table "product-list" should have "Jeans" at (1,1) in the body
 And the table "product-list" should have "Boot Cut" at (1,2) in the body
 And the table "product-list" should have "32x34" at (1,3) in the body
 And the table "product-list" should have "13.00" at (1,4) in the body
 And the table "product-list" should have "3.00" at (1,5) in the body
 And the table "product-list" should have "511420" at (1,6) in the body
```

To this: 

```Gherkin
Then the table "product-list" should have the following values:
 | Name       | Jeans     |
 | Style      | Boot Cut  | 
 | Size       | 32x34     |
 | Price      | 13.00     |
 | Sale Price | 3.00      | 
 | SKU        | 511420    | 
```

### CODE REVIEW

**Comment by medology-bugman** [comment]
Paging @atroppe to review the front-end work in this pull request!
Paging @johnnie502 to review the back-end work in this pull request!
Paging @RedEnchilada to review the testing work in this pull request!

File: features/TableContext.feature [hunk 1]
```diff
@@ -3,23 +3,30 @@ Feature: Table Context
...
-      And the table "population-table" should have "1,377,672,822" at (1,2) in the body
-      And the table "population-table" should have "2,994,159,700" at (1,2) in the footer
+      And the table "population-table" should have "1,341,335,152" at (1,4) in the body
+      And the table "population-table" should have "2,876,333,427" at (1,4) in the footer
+
+  Scenario: Developer Can Test for Row Values in the Table
+      Then the table "population-table" should have the following values:
+        | Country           | India       |
+        | Female Population | 592,067,546 |
+
```
Comment (RedEnchilada) [inline features/TableContext.feature]: Please add tests to this feature file (at least for the new step you've added) that ensure the steps fail as expected for each possible outcome. An example can be found on https://github.com/Medology/FlexibleMink/blob/master/features/FlexibleContext/assertLinesInOrder.feature using the following steps:

```gherkin
When I assert that blah
Then the assertion should throw a blah
And the assertion should fail with the message "blah"
```

The new step appears to have the following failure conditions:
- The named table does not exist
- The supplied headers don't exist on the table
- None of the rows match the expected data

Each of these should have a test to ensure they fail properly.

- [ ] Done
- [ ] Approved

**Comment by RedEnchilada** [comment]
!reassign testing

**Comment by medology-bugman** [comment]
Paging @joshmedeski to review the testing work in this pull request!

**Pull request merged**